rotkehlchen/chain/ethereum/uniswap/typing.py

```diff
@@ -65,10 +118,314 @@ def serialize(self) -> Dict[str, Any]:
+            notes=trade.notes,
+        )
+
+    def get_trade_from_ammtrade(self) -> Trade:
+        """Turns an AMMTrade to a Trade
+        """
+        # Get `fee_currency` symbol and class to instantiate
+        fc_class: Union[Type[Asset], Type[FakeAsset]]
+        if self.trade_type == TradeType.SELL:
+            fc_identifier = self.base_asset.symbol
+            fc_class = (
+                Asset
+                if isinstance(self.base_asset, EthereumToken)
+                else FakeAsset
+            )
+        else:
+            fc_identifier = self.quote_asset.symbol
+            fc_class = (
+                Asset
+                if isinstance(self.quote_asset, EthereumToken)
+                else FakeAsset
+            )
+        return Trade(
+            timestamp=self.timestamp,
+            location=self.location,
+            pair=TradePair(self.pair),
+            trade_type=self.trade_type,
+            amount=self.amount,
+            rate=self.rate,
+            fee=self.fee,
+            fee_currency=fc_class(identifier=fc_identifier),  # type: ignore
+            link=self.tx_hash,
+            notes=self.notes,
+            custom_identifier=self.identifier,
+        )
+
+    @property
+    def identifier(self) -> TradeID:
+        """Formulates a unique identifier for the trade to become the DB primary key
+
+        Current `trades` table and `Trade` class (from `data_structures.py`)
+        do not support/are not compatible with:
+          - Storing the ethereum address that did the trade.
+          - Unknown ethereum tokens, and both `pair` assets are expected to be
+          known (i.e. <Asset>).
+
+        For the reasons above, a custom trade identifier that contains the
+        the previous data.
+        """
+        partial_id = (
+            str(self.location) +
+            str(self.timestamp) +
+            str(self.trade_type) +
+            self.pair +
+            str(self.amount) +
+            str(self.rate) +
+            self.link
+        )
+        ba_, qa_ = self.base_asset, self.quote_asset
+        is_ba_unknown = 0 if isinstance(ba_, EthereumToken) else 1
+        is_qa_unknown = 0 if isinstance(qa_, EthereumToken) else 1
+
+        return TradeID(
+            f'{self.address}_'
+            f'{is_ba_unknown}_{ba_.ethereum_address}_{ba_.name}_{ba_.decimals}_'
+            f'{is_qa_unknown}_{qa_.ethereum_address}_{qa_.name}_{qa_.decimals}_'
+            f'{hash_id(partial_id)}',
+        )
+
+    @property
+    def link(self) -> str:
+        return self.tx_hash
+
+    @property
+    def pair(self) -> str:
+        return f'{self.base_asset.symbol}_{self.quote_asset.symbol}'
+
+    def serialize(self) -> Dict[str, Any]:
+        """Serialize the trade into a dict matching Trade serialization from
+        `data_structures.py` and adding: address, base_asset, quote_asset.
+        """
+        unk_asset_keys = ('ethereum_address', 'name', 'symbol')
```

Comment:
unknown_xxx. Please don't use abbreviations that make the code hard to understand